src/Google.Api.CommonProtos/TypeExtensions/Date.cs

```diff
@@ -0,0 +1,99 @@
+﻿/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file or at
+ * https://developers.google.com/open-source/licenses/bsd
+ */
+
+using System;
+
+namespace Google.Type
+{
+    /// <summary>
+    /// Extensions to <see cref="Date"/>.
+    /// </summary>
+    public partial class Date
+    {
+        /// <summary>
+        /// Converts <see cref="Date"/> to <see cref="DateTime"/>.
+        /// </summary>
+        /// <returns>The converted <see cref="DateTime"/> with no time specified.</returns>
+        /// <exception cref="InvalidOperationException">Thrown when <see cref="Date.Year"/>, <see cref="Date.Month"/>, and/or <see cref="Date.Day"/> are not within the valid range.</exception>
+        public DateTime ToDateTime()
+        {
+            try
+            {
+                return new DateTime(Year, Month, Day);
+            }
+            catch (ArgumentOutOfRangeException)
+            {
+                throw new InvalidOperationException($"Cannot convert to {nameof(DateTime)} because {nameof(Year)}, {nameof(Month)}, and/or {nameof(Day)} must be within the valid range for a {nameof(DateTime)}.");
+            }
+        }
+
+        /// <summary>
+        /// Converts <see cref="Date"/> to <see cref="DateTimeOffset"/>.
+        /// </summary>
+        /// <returns>The converted <see cref="DateTimeOffset"/> with no time specified and an Offset of <see cref="TimeSpan.Zero"/>.</returns>
```

Comment:
What Kind will the DateTime have? I don't know whether it would be better to leave it as Unspecified or make it Utc in this case. (Again, let's use "midnight".)

Reply:
@jskeet We decided to go with `Unspecified` for now since `Date` has no notion of time. We can definitely specify that the time is midnight though.